README.md

```diff
@@ -1591,6 +1593,66 @@ const { id: txId } = await io.requestPrimaryName({
 });
 ```
 
+#### `redelegateStake({ target, source, stakeQty, vaultId })`
+
+Redelegates the stake of a specific address to a new gateway. Vault ID may be optionally included in order to redelegate from an existing withdrawal vault. The redelegation fee is calculated based on the fee rate and the stake amount. Users are allowed one free redelegation every seven epochs. Each additional redelegation beyond the free redelegation will increase the fee by 10%, capping at a 60% redelegation fee.
+
+e.g: If 1000 mIO is redelegated and the fee rate is 10%, the fee will be 100 mIO. Resulting in 900 mIO being redelegated to the new gateway and 100 mIO being deducted back to the protocol balance.
+
+```typescript
+const io = IO.init({ signer: new ArweaveSigner(jwk) });
+
+const { id: txId } = await io.redelegateStake({
+  target: 't4Xr0_J4Iurt7caNST02cMotaz2FIbWQ4Kbj616RHl3',
+  source: 'HwFceQaMQnOBgKDpnFqCqgwKwEU5LBme1oXRuQOWSRA',
+  stakeQty: new IOToken(1000).toMIO(),
+  vaultId: 'fDrr0_J4Iurt7caNST02cMotaz2FIbWQ4Kcj616RHl3',
+});
+```
+
+<details>
+  <summary>Output</summary>
+
+```json
+{
+  "sourceGateway": {
+    // ...source gateway details
+  },
+  "targetGateway": {
```

Comment:
do we return the full gateway here? should we return just the addresses?

Reply:
more concerned about the Handler returning `vaults` and `delegates` with large state. let's make sure those at least are removed from the handler call

Reply:
another note - for write messages, we return the ID of the response, not the entire result object. so this actually is not the correct output for this API, can remove.

the above comment still applies about what the handler returns. i think just the addresses are suffficent.

Reply:
removed the output here, to match the other write actions:
[28edd4e](https://github.com/ar-io/ar-io-sdk/pull/273/commits/28edd4e74ca76c270f3bce4a62434aa49335dbdb)